src/read_utils.rs

```diff
@@ -214,18 +239,29 @@ where
     let file_name = path.file_name();
     let basename = file_name.map_or(path.to_string_lossy(), |b| b.to_string_lossy());
 
-    match reader {
-        None => {
+    match (reader, required_file) {
+        (None, false) => {
             info!("Skipping {}", basename);
             Ok(vec![])
         }
-        Some(reader) => {
+        (None, true) => {
+            bail!("file {:?} not found", path)
```

Comment:
This is not entirely skipping all errors since it can still create reading errors. I'm wondering if the name should be more something like `read_object_loose()`, `read_objects_weak()`, or `read_opt_objects()`? Personally, I'd prefer the first option.

Reply:
done in 8bbd58ef72aebda37fcb331abc375eb89ebd9df3